src/Stratis.Bitcoin/AsyncWork/IAsyncProvider.cs

```diff
@@ -127,6 +130,11 @@ public interface IAsyncProvider
         /// </summary>
         ISignals Signals { get; }
 
+        /// <summary>
+        /// Expose NodeLifetime as we generally need this with the AsyncProvider and saves injecting this as well.
+        /// </summary>
+        INodeLifetime NodeLifetime { get; }
```

Comment:
@ianadavies can you revert this change? We don't want to add to core components at this time. There is an outstanding task for a while now for us to consolidate often used components into a `ICoreComponent` of sorts.

Reply:
Done